core/trino-main/src/main/java/io/trino/sql/planner/IrExpressionInterpreter.java

```diff
@@ -757,6 +728,18 @@ protected Object visitCall(Call node, Object context)
             return functionInvoker.invoke(resolvedFunction, connectorSession, argumentValues);
         }
 
+        private Object processNegation(Call negation, Object context)
+        {
+            Object value = processWithExceptionHandling(negation.arguments().getFirst(), context);
+
+            return switch (value) {
+                case Call inner when inner.function().getName().getFunctionName().equals(mangleOperatorName(NEGATION)) -> inner.arguments().getFirst(); // double negation
+                case Expression inner -> new Call(negation.function(), ImmutableList.of(inner));
+                case null -> null;
```

Comment:
I've gotten the habbit of putting `null` firtst so I don't have to worry about it